client/src/components/SchedulingsView.js

```diff
@@ -0,0 +1,295 @@
+                            <Card.Description>Topic: {scheduling.topic}</Card.Description>
+                            <Card.Description>Time: {scheduling.time[0].day} from {timeSlotOptions[scheduling.time[0].time/100]}</Card.Description>
+                            <Card.Description>Your Note: {scheduling.note}</Card.Description>
+                            <br />
+                        </Card.Content>
+                            {this.buttonDisplay(scheduling)}
+                    </Card>
+                </Grid.Column>
+            </Grid.Row>
+            </Grid>
+        )
+    }
+
+    buttonDisplay(scheduling) {
+        if (this.props.activeSet === 'unconfirmed') {
+            return (
+                <Button
+                    fluid
+                    negative
+                    requestid={scheduling._id}
+                    newstatus={'Rejected'}
+                    onClick={this.handleClick.bind(this)}
+                >
+                    Cancel meeting
+                </Button>
+            )
+        } else if (this.props.activeSet === 'confirmed') {
+            return(
+                <>
+                <Button 
+                    color='blue' 
+                    as='a'
+                    href={scheduling.zoomLink} 
+                    target="_blank"
+                    rel="noopener noreferrer"
+                >
+                    Join Video Call
+                </Button>
+                <Button.Group>
+                    <Button 
+                        positive
+                        requestid={scheduling._id}
+                        newstatus={'Completed'}
+                        onClick={this.handleClick.bind(this)}
+                    >
+                        Mark as completed!
+                    </Button>
+                    <Button
+                        negative
+                        requestid={scheduling._id}
+                        newstatus={'Rejected'}
+                        onClick={this.handleClick.bind(this)}
+                    >
+                        Cancel meeting
+                    </Button>
+                </Button.Group>
+                </>
+            )
+        }
+    }
+
+    async handleClick(e) {
+        let schedulings = await makeCall({
+                requestId: e.currentTarget.getAttribute('requestid'),
+                newStatus: e.currentTarget.getAttribute('newstatus')
+            }, '/request/updateScheduling/' + this.props.userId + '/' + this.props.userRole + '/' + this.props.timeOffset, 'patch');
+        if (!schedulings || schedulings.error) {
+            swal({
+                title: "Error!",
+                text: "There was an error updating this request, please try again.",
+                icon: "error",
+            });
+        } else {
+            swal({
+                title: "Done!",
+                text: "Successfully updated this request!",
+                icon: "success"
+            })
```

Comment:
since `swal` returns a promise, anything that needs to happen synchronously after a swal should be in then block after like `swal({...}).then(() => {...})`